chainercv/datasets/siamese_dataset.py

```diff
@@ -57,6 +61,8 @@ def __init__(self, dataset_0, dataset_1,
         self._length = length
 
         if pos_ratio is not None:
+            if labels_1 is None and  dataset_0 is dataset_1:
+                labels_1 = labels_0
             if labels_0 is None:
                 labels_0 = [example[1] for example in dataset_0]
             if labels_1 is None:
```

Comment:
In the case of `labels_0 is None and labels_1 is None`, we can use `[example[1] for example in dataset_0]` for `lablels_1`. Also, the computation of `self._label_to_index_1` can be skipped.